server/src/main/java/org/elasticsearch/action/admin/indices/create/AutoCreateAction.java

```diff
@@ -84,4 +107,21 @@ protected ClusterBlockException checkBlock(CreateIndexRequest request, ClusterSt
         }
     }
 
+    static DataStreamTemplate resolveAutoCreateDataStream(CreateIndexRequest request, Metadata metadata) {
+        String v2Template = MetadataIndexTemplateService.findV2Template(metadata, request.index(), false);
+        if (v2Template != null && resolvePreferV2Templates(request)) {
+            IndexTemplateV2 indexTemplateV2 = metadata.templatesV2().get(v2Template);
+            if (indexTemplateV2.getDataStreamTemplate() != null) {
+                return indexTemplateV2.getDataStreamTemplate();
+            }
+        }
+
+        return null;
+    }
+
+    private static boolean resolvePreferV2Templates(CreateIndexRequest request) {
+        return request.preferV2Templates() == null ?
+            IndexMetadata.PREFER_V2_TEMPLATES_SETTING.get(request.settings()) : request.preferV2Templates();
```

Comment:
True, the settings in the request are always empty.  However via this way we can ensure that we always rely on prefer_v2_templates default, also when this change gets backported to 7.x.
Should we use `Settings.EMPTY` here instead of `request.settings()`?

Reply:
Sure, that makes the expectation a bit clearer. Probably a good idea to add a comment too.